include/swift/SIL/TypeLowering.h

```diff
@@ -528,7 +528,9 @@ class TypeConverter {
   llvm::DenseMap<OverrideKey, SILConstantInfo> ConstantOverrideTypes;
 
   llvm::DenseMap<AnyFunctionRef, CaptureInfo> LoweredCaptures;
-  
+
+  llvm::DenseMap<SILType, size_t> TypeFields;
```

Comment:
Everywhere else we use 'unsigned' to count the number of fields. I don't know if you'll want a struct with more than 4 billion members :-)